src/bootstrap/src/utils/shared_helpers.rs

```diff
@@ -113,3 +113,14 @@ pub fn parse_value_from_args<'a>(args: &'a [OsString], key: &str) -> Option<&'a
 
     None
 }
+
+// Flags used to compile all crates in `compiler/`.
+pub const FLAGS_FOR_RUSTC: &[&'static str] = &[
+    "-Wrustc::internal",
+    "-Drustc::symbol_intern_string_literal",
+    // FIXME(edition_2024): Change this to `-Wrust_2024_idioms` when all of the
+    // individual lints are satisfied.
+    "-Wkeyword_idents_2024",
+    "-Wunreachable_pub",
+    "-Wunsafe_op_in_unsafe_fn",
+];
```

Comment:
It seems to me these should be in a `[workspace.lints]` table, although I think you still have to opt-in `[lints] workspace = true` in each crate.

Reply:
In every `compiler/` crate? Ugh, that's the kind of thing I'm trying to get rid of in this PR :(

Reply:
It would be a one-time, one-line addition to each `Cargo.toml`, and it's a more principled way to control lints. IMO we should only use `RUSTFLAGS` (or the like) when Cargo can't do it at all.

It might even speed up the build to avoid the extra lints on out-of-tree dependencies! e.g. right now in a verbose build, I see "warning: prefer `FxHashMap` over `HashMap`" lints on a bunch of dependencies, where it's not actually applicable. Those are masked by cap-lints, but better not to run them at all.

Reply:
#138084 implements this. Unfortunately it doesn't work for rustc-specific lints like `rustc::internal`. Still, better than nothing, maybe.